underscore.js

```diff
@@ -1008,9 +1008,10 @@
   };
 
   // An internal function for creating assigner functions.
-  var createAssigner = function(keysFunc, undefinedOnly) {
+  var createAssigner = function(keysFunc, defaults) {
     return function(obj) {
       var length = arguments.length;
+      if (defaults) obj = Object(obj);
```

Comment:
`Object.assign` throws errors for this case. Not if we want to stray too far from that.

It makes sense to me in defaults. If you have nothing, and want to default it to `{a: 1}`, `{a: 1}` should be returned.